### DISCUSSION TITLE
Realtime subscribe by query

### DISCUSSION THREAD

**Discussion by alitvinenko** [appwrite/appwrite#3055] [Ideas]
Do you plan to create the ability to subscribe to changes to any documents by query?

OR

It may be possible to implement the possibility of receiving changes to the documents of the collection in real time with the collection permission set to the "document level"?
To receive changes only to documents for which you have read permission.

**Comment by gewenyu99** [comment]
I believe we have plans for more granular events in Appwrite. @Meldiron can you confirm?

**Reply by stnguyen90** [reply]
Related [comment from Torsten](https://github.com/appwrite/appwrite/issues/1625#issuecomment-924368127):

> We were thinking about extending the events to the way we are managing channels with realtime.
> 
> https://appwrite.io/docs/realtime#channels
> 
> This way you would have more control. Not sure if this would fix the create/update problem, but it lets you narrow down for what resources functions are triggered in the first place.

**Comment by eldadfux** [comment]
@alitvinenko you can do that today, all you need is to subscribe to all changes in the relevant collections. Appwrite will only notify you on events where you have read permission access to view the document.

**Reply by alitvinenko** [reply]
Yeah! Thank you!

I solved my problem. I tried subscribe on real time updates with JWT token, and It's not worked. With session cookies or/and local Storage data all is working!

This is discuss can close.

**Comment by stnguyen90** [comment]
Related [issue](https://github.com/appwrite/appwrite/issues/2490)
